megamek/src/megamek/client/ui/swing/ReplacePlayersDialog.java

```diff
@@ -31,154 +34,316 @@
     private final ClientGUI clientGui;
-    
+
     /** Convenience field for clientGui.getClient().getGame(). */
     private final Game game;
-    
-    /** The list of displayed ghost players */
-    private Set<Player> ghostPlayers;
-    
+
+    /** The id ordered list of displayed ghosts and bots players */
+    private List<Player> ghostAndBotPlayers;
+
     /** Maps a ghost player to the combobox that sets its replacement */
-    private Map<Player, JComboBox<String>> playerChoosers = new HashMap<>();
-    
+    private Map<Player, JComboBox<String>> ghostChoosers = new HashMap<>();
+
+    /** Maps a bot player to the combobox that allows it to be edited or kicked */
+    private Map<Player, JComboBox<String>> localBotChoosers = new HashMap<>();
+    /** Maps a bot player to the combobox that allows it to be kicked */
+    private Map<Player, JComboBox<String>> remoteBotChoosers = new HashMap<>();
+
     /** Maps a ghost player to the config button for the bot settings */
     private Map<Player, JButton> configButtons = new HashMap<>();
-    
+
     /** Maps a ghost player to bot settings chosen for it */
     private Map<Player, BehaviorSettings> botConfigs = new HashMap<>();
+    private final static String LOCAL = Messages.getString("ReplacePlayersDialog.local");
+    private final static String REMOTE = Messages.getString("ReplacePlayersDialog.remote");
 
-    protected ReplacePlayersDialog(JFrame frame, ClientGUI cg) {
+    protected ReplacePlayersDialog(JFrame frame, ClientGUI clientGUI) {
         super(frame, "ReplacePlayersDialog", "ReplacePlayersDialog.title");
-        clientGui = cg;
+        this.clientGui = clientGUI;
         game = clientGui.getClient().getGame();
-        ghostPlayers = game.getPlayersVector().stream()
-                .filter(Player::isGhost).collect(Collectors.toSet());
-        // Add a new Princess behavior for each ghost. It will be overwritten by savegame behaviors
-        ghostPlayers.forEach(p -> botConfigs.put(p, new BehaviorSettings()));
+        refreshPlayers();
         initialize();
     }
-    
+
     @Override
     protected void initialize() {
         super.initialize();
         adaptToGUIScale();
         try {
             finalizeInitialization();
         } catch (Exception ex) {
-            LogManager.getLogger().error("Error finalizing the ReplacePlayersDialog. Returning the created dialog, but this is likely to cause some oddities.", ex);
+            LogManager.getLogger().error("Error finalizing the ManagePlayersDialog. Returning the created dialog, but this is likely to cause some oddities.", ex);
         }
     }
 
+    protected void refreshPlayers() {
+        ghostAndBotPlayers = game.getPlayersVector().stream()
+                .filter(c -> c.isGhost() || c.isBot()).collect(Collectors.toList());
+        ghostAndBotPlayers.sort(Comparator.comparingInt(Player::getId));
+        ghostAndBotPlayers.forEach(p -> botConfigs.put(p, new BehaviorSettings()));

```

Comment:
Can we please chain these calls?

Also, instead of c for client and p for player can we use client and player to prevent misreads?